node/websocket/websocket_test.go

```diff
@@ -0,0 +1,84 @@
+package websocket_test
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/INFURA/go-ethlibs/node/websocket"
+)
+
+func getRopstenConnection(t *testing.T, ctx context.Context) websocket.Connection {
+	// These test require a ropsten websocket URL to test with, for example ws://localhost:8546 or wss://ropsten.infura.io/ws/v3/:YOUR_PROJECT_ID
+	url := os.Getenv("INFURA_TEST_ROPSTEN_WS_URL")
+	if url == "" {
+		t.Skip("INFURA_TEST_ROPSTEN_WS_URL not set, skipping test.  Set to a valid websocket URL to execute this test.")
+	}
+
+	conn, err := websocket.NewConnection(ctx, url)
+	require.NoError(t, err, "creating websocket connection should not fail")
+	return conn
+}
+
+func TestConnection_FutureBlockByNumber(t *testing.T) {
+	ctx := context.TODO()
```

Comment:
(optional) I think is supposed to be used if you're intending to go back and replace it with a specific context later (e.g. during a refactor). When in doubt, `context.Background()` is the way to go.

Reply:
Hah actually I set it to TODO as a reminder to myself since I wanted to see if `testing.T` had a context we could use.  turns out it (briefly) did but was removed before Go 1.8 went live: https://github.com/golang/go/issues/18199

Doesn't look like that's going to be re-added anytime soon so I'll switch to .Background.

Reply:
Ha, nice. `TODO()` is also practically deprecated too. -_- There were plans to add linter tooling for detecting it during vet etc, but it never happened and the issue got closed.

Reply:
Switched in beda050.